packages/esm-service-queues-app/src/queue-services/queue-service-form.workspace.tsx

```diff
@@ -1,180 +1,174 @@
+          kind: 'error',
+          isLowContrast: false,
+          subtitle: error?.message,
+        });
+      });
+  };
 
   return (
-    <Form onSubmit={createQueue} className={styles.form}>
-      <Stack gap={4} className={styles.grid}>
+    <Form onSubmit={handleSubmit(createQueue)} className={styles.form}>
+      <Stack gap={5} className={styles.grid}>
         <Column>
           <Layer className={styles.input}>
-            <TextInput
-              id="queueName"
-              invalidText="Required"
-              labelText={t('queueName', 'Queue name')}
-              onChange={(event) => setQueueName(event.target.value)}
-              value={queueName}
-            />
-            {isMissingName && (
-              <section>
-                {/* TODO: Use a zod schema instead of this */}
-                <InlineNotification
-                  style={{ margin: '0', minWidth: '100%' }}
-                  kind="error"
-                  lowContrast
-                  title={t('missingQueueName', 'Missing queue name')}
-                  subtitle={t('addQueueName', 'Please add a queue name')}
+            <Controller
+              name="queueName"
+              control={control}
+              render={({ field }) => (
+                <TextInput
+                  {...field}
+                  id="queueName"
+                  invalidText={errors.queueName?.message}
+                  invalid={!!errors.queueName}
+                  labelText={t('queueName', 'Queue name')}
                 />
-              </section>
-            )}
+              )}
+            />
           </Layer>
         </Column>
 
         <Column>
           <Layer className={styles.input}>
-            <Select
-              labelText={t('selectServiceType', 'Select a service type')}
-              id="queueConcept"
-              invalidText="Required"
-              onChange={(event) => setQueueConcept(event.target.value)}
-              value={queueConcept}>
-              {!queueConcept && <SelectItem text={t('selectServiceType', 'Select a service type')} value="" />}
-              {queueConcepts.length === 0 && (
-                <SelectItem text={t('noServicesAvailable', 'No services available')} value="" />
+            <Controller
+              name="queueConcept"
+              control={control}
+              render={({ field }) => (
+                <Select
+                  {...field}
+                  labelText={t('selectServiceType', 'Select a service type')}
+                  id="queueConcept"
+                  invalid={!!errors?.queueConcept}
+                  invalidText={errors?.queueConcept?.message}>
+                  {!field.value && <SelectItem text={t('selectServiceType', 'Select a service type')} value="" />}
+                  {queueConcepts.length === 0 && (
+                    <SelectItem text={t('noServicesAvailable', 'No services available')} value="" />
+                  )}
```

Comment:
```suggestion
                  {queueConcepts.length === 0 && (
                    <SelectItem text={t('noServicesAvailable', 'No services available')} value="" />
                  )}
```

I think we can do without this block. Probably better to make the input disabled if there are no services to display and show some helper text below the input.